src/onboarding/create-utils.js

```diff
@@ -39,3 +40,27 @@ export function prepareTransactionCreatorFromAbi(abi, toAddress) {
     }
   }
 }
+
+export async function getGasPrice() {
+  const web3 = getWeb3(web3Providers.default)
+  try {
+    const response = await fetch(
+      'https://ethgasstation.info/json/ethgasAPI.json',
+      {
+        method: 'GET',
+        mode: 'cors',
+        cache: 'no-cache',
+      }
+    )
+    const jsonResponse = await response.json()
+    const priceInWei = web3.utils.toWei(
+      jsonResponse.gasPriceRange[40].toString(), // using '40' as a safe price, this should be set via user input
+      'gwei'
+    )
+    return priceInWei
+  } catch (e) {
+    log('Error fetching gas price: ', e)
+  }
+  const priceInWei = web3.utils.toWei('2', 'gwei') // If we can't fetch a price default to 2 gwei
```

Comment:
In relation to above, I'd make this a function input.

The user of the function can suggest what a reasonable minimum is (e.g. 3 gwei for onboarding tx), and if the current suggested price is lower, or can't be fetched, it defaults to that minimum.